### PR TITLE
Add call to partial import endpoint.

### PR DESCRIPTION
**Pull request by ssilvert**

### CODE REVIEW

File: src/defs/realmRepresentation.ts [hunk 1]
```diff
@@ -116,3 +116,21 @@ export default interface RealmRepresentation {
   verifyEmail?: boolean;
   waitIncrementSeconds?: number;
 }
+
+export type PartialImportRealmRepresentation = RealmRepresentation & {
```
Comment (jonkoops) [inline src/defs/realmRepresentation.ts:120]: According to the [documentation](https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_partialimport) this should be named `PartialImportRepresentation`.

```suggestion
export type PartialImportRepresentation = RealmRepresentation & {
```

It also looks to me that [`PartialImportRepresentation`](https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_partialimportrepresentation) doesn't have all of the fields that [`RealmRepresentation`](https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_realmrepresentation) has. Perhaps we should not be extending this existing type, instead adding all the fields explicitly.

File: src/resources/realms.ts [hunk 2]
```diff
@@ -53,6 +56,19 @@ export class Realms extends Resource {
     urlParamKeys: ['realm'],
   });
 
+  public partialImport = this.makeRequest<
+    {
+      realm: string;
+      realmRep: PartialImportRealmRepresentation;
```
Comment (jonkoops) [inline src/resources/realms.ts:62]: In the [API docs](https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_partialimport) this is simply called `rep`.
```suggestion
      rep: PartialImportRepresentation;
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -11,8 +11,9 @@
     "clean": "rimraf lib",
     "lint": "tslint --force --format verbose \"src/**/*.ts\" \"test/*\"",
     "build": "npm run clean && tsc -p ./tsconfig.release.json --pretty",
-    "test": "DEBUG=kc-admin mocha --require ts-node/register --recursive \"test/**/*.spec.ts\"",
-    "test:grep": "DEBUG=kc-admin mocha --require ts-node/register",
+    "test": "mocha --require ts-node/register --recursive \"test/**/*.spec.ts\"",
+    "test:subset": "mocha --require ts-node/register --recursive --",
```
[the remark this answers is not in the document]
Comment (ssilvert) [reply package.json:15]: It's useful if you want to run a single test by itself.

File: src/defs/realmRepresentation.ts (hunk 1 above)
Comment (ssilvert) [reply src/defs/realmRepresentation.ts:120]: I don't agree with the shorter name.  When you do the import, the json often contains the entire realm representation that was exported from the Keycloak server at startup.  So even though we don't care about all the fields, it is legal for them to be there.

File: src/resources/realms.ts [hunk 4]
```diff
@@ -53,6 +56,19 @@ export class Realms extends Resource {
...
+    {
+      realm: string;
+      rep: PartialImportRepresentation;
+    },
+    PartialImportResponse
+  >({
+    method: 'POST',
+    path: '/{realm}/partialImport',
+    urlParamKeys: ['realm'],
+    payloadKey: 'realmRep',
```
Comment (ssilvert) [inline src/resources/realms.ts:69]:
```suggestion
    payloadKey: 'rep',
```

File: src/resources/realms.ts [hunk 5]
```diff
@@ -53,6 +56,19 @@ export class Realms extends Resource {
     urlParamKeys: ['realm'],
   });
 
+  public partialImport = this.makeRequest<
+    {
+      realm: string;
+      rep: PartialImportRepresentation;
```
Comment (ssilvert) [inline src/resources/realms.ts:62]:
```suggestion
      rep: PartialImportRealmRepresentation;
```

File: test/realms.spec.ts [hunk 6]
```diff
@@ -66,6 +67,34 @@ describe('Realms', () => {
...
+          composite: false,
+        },
+      ],
+    },
+  };
+
+  it('does partial import', async () => {
+    const result = await kcAdminClient.realms.partialImport({
+      realm: currentRealmName,
+      realmRep: roleToImport,
```
Comment (ssilvert) [inline test/realms.spec.ts:87]:
```suggestion
      rep: roleToImport,
```

File: src/defs/realmRepresentation.ts (hunk 1 above)
Comment (jonkoops) [reply src/defs/realmRepresentation.ts:120]: I'd rather follow what the API documentation provides as that is what we do for all other pieces, but if it's really that important I guess we can make an exception.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**